dotnet/CustomerServiceAgent/docs/setup/02-entra-id-setup.md

```diff
@@ -0,0 +1,329 @@
+
+### Step 6: Create Blueprint
+
+1. Navigate to **Microsoft Entra ID** → **Identity Governance** (or search for "Agent Identities")
+2. Select **Agent Identity Blueprints**
+3. Click **New blueprint**
+4. Configure:
+   - **Name:** `CustomerServiceAgentBlueprint`
+   - **Description:** `Blueprint for customer service agents`
+5. Click **Create**
+6. Note the **Blueprint ID**
+
+### Step 7: Create Autonomous Agent Identity
+
+1. In the blueprint, click **Create agent identity**
+2. **Type:** Autonomous Agent
+3. Configure:
+   - **Name:** `CustomerServiceAutonomousAgent`
+   - **Description:** `Autonomous agent for read operations`
+4. Click **Create**
+5. Note the **Agent Identity ID**
+
+### Step 8: Create Agent User Identity
+
+1. In the blueprint, click **Create agent identity**
+2. **Type:** Agent User
+3. Configure:
+   - **Name:** `CustomerServiceAgentUser`
+   - **Description:** `Agent user for write operations`
+   - **Associated User:** Select or create a service account (e.g., `csr-agent@yourdomain.com`)
+4. Click **Create**
+5. Note the **Agent User Identity ID**
+
+---
+
+## Part 4: Update Configuration (5 minutes)
+
+### Step 9: Update Orchestrator appsettings.json
+
+Edit `src/AgentOrchestrator/appsettings.json`:
+
+```json
+{
+  "AzureAd": {
+    "Instance": "https://login.microsoftonline.com/",
+    "TenantId": "YOUR_TENANT_ID",
+    "ClientId": "YOUR_ORCHESTRATOR_CLIENT_ID",
+    "ClientSecret": "YOUR_ORCHESTRATOR_CLIENT_SECRET"
+  },
+  "AgentIdentities": {
+    "AgentIdentity": "YOUR_AUTONOMOUS_AGENT_ID",
+    "AgentUserId": "YOUR_AGENT_USER_ID"
+  },
+  "Services": {
+    "OrderService": "https://localhost:7001",
+    "CrmService": "https://localhost:7002",
+    "ShippingService": "https://localhost:7003",
+    "EmailService": "https://localhost:7004"
+  }
+}
+```
+
+### Step 10: Update Downstream Service appsettings.json
+
+For **each service** (Order, CRM, Shipping, Email):
+
+Edit `src/DownstreamServices/[ServiceName]/appsettings.json`:
+
+```json
+{
+  "AzureAd": {
+    "Instance": "https://login.microsoftonline.com/",
+    "TenantId": "YOUR_TENANT_ID",
+    "ClientId": "YOUR_SERVICE_CLIENT_ID",
+    "Audience": "api://YOUR_SERVICE_CLIENT_ID"
+  }
+}
+```
+
+**Example for Order Service:**
+```json
+{
+  "AzureAd": {
+    "Instance": "https://login.microsoftonline.com/",
+    "TenantId": "a1b2c3d4-e5f6-g7h8-i9j0-k1l2m3n4o5p6",
+    "ClientId": "11111111-2222-3333-4444-555555555555",
+    "Audience": "api://11111111-2222-3333-4444-555555555555"
+  }
+}
+```
+
+### Step 11: Enable Authentication in Code
+
+#### Orchestrator (src/AgentOrchestrator/Program.cs)
+
+Uncomment the authentication configuration:
+
+```csharp
+// Use this pattern:
+builder.Services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
+    .AddMicrosoftIdentityWebApi(builder.Configuration, "AzureAd")
+        .EnableTokenAcquisitionToCallDownstreamApi()
```

Comment:
Let's move this to builder.Services.AddTokenAcquisition(true).AddInMemoryTokenCaches()l